js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+}
+
+// Alignment options for the canvas
+export enum Alignment {
+  Center       = 'center',
+  TopLeft      = 'topLeft',
+  TopCenter    = 'topCenter',
+  TopRight     = 'topRight',
+  CenterLeft   = 'centerLeft',
+  CenterRight  = 'centerRight',
+  BottomLeft   = 'bottomLeft',
+  BottomCenter = 'bottomCenter',
+  BottomRight  = 'bottomRight'
+}
+
+// Interface for the Layout static method contructor
+export interface LayoutParameters {
+  fit: Fit,
+  alignment: Alignment,
+  minX: number,
+  minY: number,
+  maxX: number,
+  maxY: number
+}
+
+// Alignment options for Rive animations in a HTML canvas
+export class Layout {
+  public fit: Fit;
+  public alignment: Alignment;
+  public minX: number;
+  public minY: number;
+  public maxX: number;
+  public maxY: number;
+
+  constructor(
+    fit: Fit = Fit.None,
+    alignment: Alignment = Alignment.Center,
+    minX: number = 0,
+    minY: number = 0,
+    maxX: number = 0,
+    maxY: number = 0
+  ) {
+    this.fit = fit;
+    this.alignment = alignment;
+    this.minX = minX;
+    this.minY = minY;
+    this.maxX = maxX;
+    this.maxY = maxY;
+  }
+
+  // Alternative constructor to build a Layout from an interface/object
+  static new({fit, alignment, minX, minY, maxX, maxY}: LayoutParameters) : Layout {
+    return new Layout(fit, alignment, minX, minY, maxX, maxY);
+  }
+
+  // Returns fit for the Wasm runtime format
+  public runtimeFit(rive: any): any {
+    switch (this.fit) {
+      case Fit.Cover:
+        return rive.Fit.cover;
+      case Fit.Contain:
+        return rive.Fit.contain;
+      case Fit.Fill:
+        return rive.Fit.fill;
+      case Fit.FitWidth:
+        return rive.Fit.fitWidth;
+      case Fit.FitHeight:
+        return rive.Fit.fitHeight;
+      case Fit.ScaleDown:
+        return rive.Fit.scaleDown;
+      case Fit.None:
+      default:
+        return rive.Fit.none;
+    }
+  }
+
+  // Returns alignment for the Wasm runtime format
+  public runtimeAlignment(rive: any): any {
+    switch (this.alignment) {
+      case Alignment.TopLeft:
+        return rive.Alignment.topLeft;
+      case Alignment.TopCenter:
+        return rive.Alignment.topCenter;
+      case Alignment.TopRight:
+        return rive.Alignment.topRight;
+      case Alignment.CenterLeft:
+        return rive.Alignment.centerLeft;
+      case Alignment.CenterRight:
+        return rive.Alignment.centerRight;
+      case Alignment.BottomLeft:
+        return rive.Alignment.bottomLeft;
+      case Alignment.BottomCenter:
+        return rive.Alignment.bottomCenter;
+      case Alignment.BottomRight:
+        return rive.Alignment.bottomRight;
+      case Alignment.Center:
+      default:
+        return rive.Alignment.center;
+    }
+  }
+}
+
+// #endregion
+
+// #region runtime
+
+// Callback type when looking for a runtime instance
+export type RuntimeCallback = (rive: typeof Runtime) => void;
+
+// Runtime singleton; use getInstance to provide a callback that returns the
```

Comment:
I would recommend JSdoc format comments for exported API to let the editor display information to the developer. This also will reduce the amount of required documentation :
```typescript
/**
 * Runtime singleton...
 */
```

Reply:
I'll read up on JSdoc